### ISSUE TITLE
Canvas.raster(): wrong output coords when sampling on same x range and y range as of source input

### ISSUE THREAD

**Issue by thuydotm** [holoviz/datashader#958]
#### ALL software version info
datashader version 0.11.1

#### Description of expected behavior and the observed behavior
Seems like there is a bug in Canvas.raster() that makes the coords of output DataArray incompatible with the source input DataArray when sampling on the same x range and y range.

#### Complete, minimal, self-contained example code that reproduces the issue

```python
import pandas as pd
import datashader as ds
from datashader.transfer_functions import shade, stack

width, height  = 800, 600
df = pd.DataFrame({'y': [-3, -1, -4, -4, 1, 3], 'x': [-5, -8, -5, -5, 0, 5]})

cvs = ds.Canvas(plot_width=width, plot_height=height, x_range=(-10, 10), y_range=(-5, 4))

line_agg = cvs.line(df, x='x', y='y')
sample_raster_agg = cvs.raster(line_agg)

img1 = shade(sample_raster_agg)
img2 = shade(line_agg)

stack(img1, img2)
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-fee18f015693> in <module>
      14 img2 = shade(line_agg)
      15 
----> 16 stack(img2, img1)

~\Anaconda3\envs\xarray-spatial\lib\site-packages\datashader\transfer_functions\__init__.py in stack(*imgs, **kwargs)
    134     imgs = xr.align(*imgs, copy=False, join='outer')
    135     with np.errstate(divide='ignore', invalid='ignore'):
--> 136         out = tz.reduce(tz.flip(op), [i.data for i in imgs])
    137     return Image(out, coords=imgs[0].coords, dims=imgs[0].dims, name=name)
    138 

~\Anaconda3\envs\xarray-spatial\lib\site-packages\toolz\functoolz.py in __call__(self, *args, **kwargs)
    301     def __call__(self, *args, **kwargs):
    302         try:
--> 303             return self._partial(*args, **kwargs)
    304         except TypeError as exc:
    305             if self._should_curry(args, kwargs, exc):

~\Anaconda3\envs\xarray-spatial\lib\site-packages\toolz\functoolz.py in flip(func, a, b)
    735     [1, 2, 3]
    736     """
--> 737     return func(b, a)
    738 
    739 

TypeError: ufunc 'over' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

```

Checking their coords:
```
>>> (sample_raster_agg.coords['y'].data == line_agg.coords['y'].data).all(),  (sample_raster_agg.coords['x'].data == line_agg.coords['x'].data).all()
(False, False)
```

**Comment by thuydotm** [comment]
The root cause here is because of the extremely small errors when calculating coordinates for the output:

![image](https://user-images.githubusercontent.com/47617499/92067236-c925b380-ed71-11ea-827f-5a7c29e36d87.png)

**Comment by jbednar** [comment]
I believe this is addressed by #1038; closing.
